build_rsas_from_etas.gs

```diff
@@ -0,0 +1,390 @@
+  var topLevelAccountId = AdsApp.currentAccount().getCustomerId();
+  var ssName = topLevelAccountName + " (" + topLevelAccountId + ") | ETA to RSA Builder";
+  
+  Logger.log("Making Google Sheet...");
+  var ss = SpreadsheetApp.create(ssName);
+  
+  // If MCC, run data process on a loop through all accounts
+  if (executionContext === 'manager_account') {
+    var accountIterator = AdsManagerApp.accounts().get();
+    while(accountIterator.hasNext()) {
+      AdsManagerApp.select(accountIterator.next());
+      
+      var accountName = AdsApp.currentAccount().getName();
+      var accountId = AdsApp.currentAccount().getCustomerId();
+      var accountNameAndId = accountName + " (" + accountId + ")";  
+      var successCount = 0;
+      
+      Logger.log("Looking in account " + accountNameAndId + " ...");
+      var assets = getAssets(accountName, accountId);
+      if (assets !== 'no_rsas') {
+        var sheet = ss.insertSheet(accountNameAndId)
+        var maxNoOfAssets = getMaxAssets(assets);
+        writeToSheet(sheet, assets, maxNoOfAssets, accountName, accountId);
+        
+        successCount++;
+      }
+      
+      else if (assets === 'no_rsas' && accountIterator.hasNext()) {
+        Logger.log("Trying next account...");
+      }
+      else if (assets === 'no_rsas' && !accountIterator.hasNext()) {
+        Logger.log("No more accounts found. Terminating script.");
+      }   
+       
+      Logger.log("--------------------------------------------------");
+    }
+    
+    // Remove first sheet, which is blank
+    ss.deleteSheet(ss.getSheets()[0]);
+    if (successCount === 0) {
+      Logger.log("No ad groups missing RSAs found. Google Sheet can be safely deleted.");
+    }
+    Logger.log("Script run complete. Google Sheet location: " + ss.getUrl());
+  }
+  
+  // If child account, process on that account only
+  else if (executionContext === 'client_account') {
+    var accountName = AdsApp.currentAccount().getName();
+    var accountId = AdsApp.currentAccount().getCustomerId();
+    var accountNameAndId = accountName + " (" + accountId + ")";
+    var sheet = ss.getSheets()[0].setName(accountNameAndId);
+    
+    var assets = getAssets(accountName, accountId);
+    if (assets === 'no_rsas') {
+      Logger.log("Terminating script. Google Sheet can be safely deleted.");
+    }
+    
+    else if (assets !== 'no_rsas') {
+      var maxNoOfAssets = getMaxAssets(assets);
+      writeToSheet(sheet, assets, maxNoOfAssets, accountName, accountId);
+    }
+    
+    Logger.log("--------------------------------------------------");
+    Logger.log("Script run complete. Google Sheet location: " + ss.getUrl()); 
+
+  }
+}
+
+// ========= UTILITY FUNCTIONS ==========================================================================================
+
+// Determine the type of account in which the script is running
+function getExecutionContext() {
+    if (typeof AdsManagerApp != "undefined") {
```

Comment:
Is this the only case when an account is set as undefined?

Reply:
`typeof AdsManagerApp` returns `undefined` if the account the script is being run in a child account (i.e. not MCC), so this checks if the script is in an MCC or not.

Also interesting to note I've used `!= "undefined"` here, when it should really be `!== undefined`